sklearn/utils/extmath.py

```diff
@@ -1232,10 +1241,30 @@ def stable_cumsum(arr, axis=None, rtol=1e-05, atol=1e-08):
     out : ndarray
         Array with the cumulative sums along the chosen axis.
     """
-    out = np.cumsum(arr, axis=axis, dtype=np.float64)
-    expected = np.sum(arr, axis=axis, dtype=np.float64)
-    if not np.allclose(
-        out.take(-1, axis=axis), expected, rtol=rtol, atol=atol, equal_nan=True
+    xp, _, device = get_namespace_and_device(arr)
+    if axis is None:
+        # np.cumsum calculates cumsum across flattened array
+        # for axis=None, which is why we flatten here.
+        # (if we don't flatten, cumulative_sum below will raise error)
+        arr = xp.reshape(arr, (-1,))
```

Comment:
What about adding:
```suggestion
        arr = xp.reshape(arr, (-1,))
        axis=0
```
Avoids the `if axis is None:` below

Reply:
Thanks, that's much nicer than what I had.